src/main/java/budgetbuddy/logic/parser/commandparsers/transactioncommandparsers/TransactionAddCommandParser.java

```diff
@@ -65,10 +68,11 @@ public TransactionAddCommand parse(String args) throws ParseException {
                 ? CommandParserUtil.parseAccount(optionalAccount.get())
                 : null;
 
-        Optional<String> optionalCategory = argMultiMap.getValue(PREFIX_CATEGORY);
-        Category category = optionalCategory.isPresent()
-                ? CommandParserUtil.parseCategory(optionalCategory.get())
-                : null;
+        List<String> categoriesList = argMultiMap.getAllValues(PREFIX_CATEGORY);
+        Set<Category> categoriesSet = new HashSet<>();
+        for (String c : categoriesList) {
+            categoriesSet.add(new Category(c));
+        }
```

Comment:
For this I think you should be doing a `CommandParserUtil.parseCategory` for each category you want to add.

Reply:
Okay, thanks! changed